tests/test_generators.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 Graz University of Technology.
+#
+# Invenio-Records-Permissions is free software; you can redistribute it
+# and/or modify it under the terms of the MIT License; see LICENSE file for
+# more details.
+
+"""Pytest configuration.
+
+See https://pytest-invenio.readthedocs.io/ for documentation on which test
+fixtures are available.
+"""
+
+import pytest
+from invenio_access.permissions import any_user, authenticated_user
+from invenio_records_permissions.generators import AnyUser, AuthenticatedUser
+
+from invenio_rdm_records.generators import IfRestricted
+
+
+@pytest.mark.parametrize("field", ['files'])
+def test_ifrestricted(field, create_record):
+    # Restricted files, permission level = then_
+    record = create_record(
+        {
+            "access": {
+                "owned_by": [{"user": 4}],
+                "record": "public",
+                "files": "restricted",
+                "grants": [
+                    {"subject": "user", "id": 1, "level": "edit"},
+                    ]
+                }
+        }
+    )
+
+    generator = IfRestricted(field=field,
+                             then_=[AuthenticatedUser()], else_=[AnyUser()])
+    assert generator.needs(record=record) == [authenticated_user]
```

Comment:
Minor: Can you add a test to see that the ``else_`` part is covered as well. Something along the lines of:

```
generator = IfRestricted(field='files', then_=[AuthenticatedUser()], else_=[AnyUser()])
asser generator.needs(record=record) == [authenticated_user]
generator = IfRestricted(field='record', then_=[AuthenticatedUser()], else_=[AnyUser()])
asser generator.needs(record=record) == [any_user]
```